website/source/assets/javascripts/configuration-builder.js

```diff
@@ -0,0 +1,133 @@
+function downloadConfiguration() {
+  var form = document.querySelector("#configuration-builder");
+  var config = "";
+
+  // Add Listener stanza
+  if (document.getElementById("include_tcp_listener").checked) {
+    config += `listener "tcp" {
+${addFieldsToStanza("listener")}}
+`;
+  }
+
+  // Add Storage stanza
+  if (document.getElementById("include_storage").checked) {
+    var backend = document.getElementById("storage").value;
+    config += `
+storage "${backend}" {
+${addFieldsToStanza("storage")}}
+`;
+  }
+
+  // Add Telemetry stanza
+  if (document.getElementById("include_telemetry").checked) {
+    var provider = document.getElementById("telemetry").value;
+    config += `
+telemetry {
+${addFieldsToStanza("telemetry")}}
+`;
+  }
+
+  // Add Seal stanza
+  if (document.getElementById("include_seal").checked) {
+    var type = document.getElementById("seal").value;
+    config += `
+seal "${type}" {
+${addFieldsToStanza("seal")}}
+`;
+  }
+
+  // Add UI stanza
+  if (document.getElementById("include_ui").checked) {
+    config += `
+ui = true`;
+    var startServerLink = document.querySelector(".start-server-link")
+    startServerLink.href = `${startServerLink.href}?tab=ui`;
+  }
+
+  config = config.replace(/([^\r])\n/g, "$1\r\n");
+  var blob = new Blob([config], {type: "text/plain;charset=utf-8"});
+  saveAs(blob, "vault-config.hcl");
+  document.querySelector(".form-actions").style.display = "none";
+  document.querySelector("#download-confirm").style.display = "block";
+  return false;
```

Comment:
Just curious, is the `return false` important?

Reply:
It was at one point because clicking the button was refreshing the page and keeping you from seeing the confirmation message. I ended up having to add `type="button"` for another bug and looks like it doesn't refresh now (should have realized why that was happening earlier). Thanks!